orchagent/portsorch.cpp

```diff
@@ -2177,22 +2179,92 @@ void PortsOrch::doPortTask(Consumer &consumer)
                 }
 
                 /* Set port serdes Pre-emphasis */
+                attr_val.clear();
                 if (fvField(i) == "preemphasis")
                 {
-                    getPortSerdesVal(fvValue(i), pre_emphasis);
+                    getPortSerdesVal(fvValue(i), attr_val);
+                    serdes_attr.insert(serdes_attr_pair(SAI_PORT_SERDES_ATTR_PREEMPHASIS, attr_val));
                 }
 
+                attr_val.clear();
                 /* Set port serdes idriver */
                 if (fvField(i) == "idriver")
                 {
-                    getPortSerdesVal(fvValue(i), idriver);
+                    getPortSerdesVal(fvValue(i), attr_val);
+                    serdes_attr.insert(serdes_attr_pair(SAI_PORT_SERDES_ATTR_IDRIVER, attr_val));
                 }
 
+                attr_val.clear();
                 /* Set port serdes ipredriver */
                 if (fvField(i) == "ipredriver")
                 {
-                    getPortSerdesVal(fvValue(i), ipredriver);
+                    getPortSerdesVal(fvValue(i), attr_val);
+                    serdes_attr.insert(serdes_attr_pair(SAI_PORT_SERDES_ATTR_IPREDRIVER, attr_val));
+                }
+
+                /* Set port serdes pre1 */
+                attr_val.clear();
+                if (fvField(i) == "pre1")
```

Comment:
Can we have these values like "pre1", "post1", "attn", etc. be macros?

Reply:
Hi @daall, The existing fields are also compared as plain strings and hence to maintain uniformity I didn't define macros. Moreover this string is just used in one place. Please let me know if you want to change these to macros or the strings should be good

Reply:
That seems reasonable to me, we can keep these as is.